### ISSUE TITLE
STF keep on trying to find for minicap.so file for android 11

### ISSUE THREAD

**Issue by Babu-SKK** [DeviceFarmer/stf#54]
**What is the issue or idea you have?**
STF keep on trying to find for minicap.so file for android 11 devices. But minicap doesn't support the Androdi 11 yet. Due to this, sometimes stf is not working properly

**Does it only happen on a specific device? Please run `adb devices -l` and paste the corresponding row.**
yes Only on Android 11 (R) devices

**Please provide the steps to reproduce the issue.**
Connect Android R device & check

**What is the expected behavior?**
It should stop the minicap.so finding/pushing after trying few times. After that it has to ignore that device & it should show as either disconnected/setup error in device list

**Do you see errors or warnings in the `stf local` output? If so, please paste them or the full log here.**

```
2020-07-30T09:02:29.639Z FTL/util:lifecycle 17530 [R38MB0AHCRN] Shutting down due to fatal error
2020-07-30T09:02:29.654Z INF/provider 8400 [*] Cleaning up device worker "R38MB0AHCRN"
2020-07-30T09:02:29.654Z ERR/provider 8400 [*] Device worker "R38MB0AHCRN" died with code 1
2020-07-30T09:02:29.654Z INF/provider 8400 [*] Restarting device worker "R38MB0AHCRN"
^[[B^[[B^[[B2020-07-30T09:02:30.625Z INF/device:support:push 17744 [R38MB0AHCRN] Sending output to "tcp://127.0.0.1:7116"
2020-07-30T09:02:30.632Z INF/device 17744 [R38MB0AHCRN] Preparing device
^[[B^[[B^[[B^[[B^[[B^[[B2020-07-30T09:02:31.246Z INF/device:support:sub 17744 [R38MB0AHCRN] Receiving input from "tcp://127.0.0.1:7114"
2020-07-30T09:02:31.248Z INF/device:support:sub 17744 [R38MB0AHCRN] Subscribing to permanent channel "*ALL"
^[[B^[[B^[[B^[[B^[[B^[[B2020-07-30T09:02:31.321Z INF/device:support:properties 17744 [R38MB0AHCRN] Loading properties
2020-07-30T09:02:31.371Z INF/device:support:sdk 17744 [R38MB0AHCRN] Supports SDK 30
2020-07-30T09:02:31.372Z INF/device:support:abi 17744 [R38MB0AHCRN] Supports ABIs arm64-v8a, armeabi-v7a, armeabi
2020-07-30T09:02:31.375Z FTL/device 17744 [R38MB0AHCRN] Setup had an error Error: At least one of these paths should exist: /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/arm64-v8a/lib/android-30/minicap.so, /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/arm64-v8a/lib/android-30/minicap.so, /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/armeabi-v7a/lib/android-30/minicap.so, /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/armeabi-v7a/lib/android-30/minicap.so, /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/armeabi/lib/android-30/minicap.so, /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/armeabi/lib/android-30/minicap.so
    at Object.module.exports.requiredMatch (/home/bob/workspace/STF/lib/util/pathutil.js:33:9)
    at /home/bob/workspace/STF/lib/units/device/resources/minicap.js:41:23
    at SerialSyrup.ParallelSyrup.invoke (/home/bob/workspace/STF/node_modules/stf-syrup/lib/parallel.js:54:24)
    at /home/bob/workspace/STF/node_modules/stf-syrup/lib/serial.js:43:33
    at tryCatch1 (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/util.js:63:19)
    at Promise$_callHandler [as _callHandler] (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/promise.js:708:13)
    at Promise$_settlePromiseFromHandler [as _settlePromiseFromHandler] (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/promise.js:724:18)
    at Promise$_settlePromiseAt [as _settlePromiseAt] (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/promise.js:896:14)
    at Promise$_fulfillPromises [as _fulfillPromises] (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/promise.js:1041:14)
    at Async$_consumeFunctionBuffer [as _consumeFunctionBuffer] (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/async.js:74:12)
    at Async$consumeFunctionBuffer (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/async.js:37:14)
    at _combinedTickCallback (internal/process/next_tick.js:132:7)
    at process._tickCallback (internal/process/next_tick.js:181:9)
2020-07-30T09:02:31.376Z FTL/util:lifecycle 17744 [R38MB0AHCRN] Shutting down due to fatal error
2020-07-30T09:02:31.386Z INF/provider 8400 [*] Cleaning up device worker "R38MB0AHCRN"
2020-07-30T09:02:31.386Z ERR/provider 8400 [*] Device worker "R38MB0AHCRN" died with code 1
2020-07-30T09:02:31.386Z INF/provider 8400 [*] Restarting device worker "R38MB0AHCRN"
^[[B^[[B^[[B2020-07-30T09:02:32.300Z INF/device:support:push 17758 [R38MB0AHCRN] Sending output to "tcp://127.0.0.1:7116"
2020-07-30T09:02:32.305Z INF/device 17758 [R38MB0AHCRN] Preparing device
^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B2020-07-30T09:02:32.946Z INF/device:support:sub 17758 [R38MB0AHCRN] Receiving input from "tcp://127.0.0.1:7114"
2020-07-30T09:02:32.948Z INF/device:support:sub 17758 [R38MB0AHCRN] Subscribing to permanent channel "*ALL"
2020-07-30T09:02:33.017Z INF/device:support:properties 17758 [R38MB0AHCRN] Loading properties
2020-07-30T09:02:33.064Z INF/device:support:sdk 17758 [R38MB0AHCRN] Supports SDK 30
2020-07-30T09:02:33.065Z INF/device:support:abi 17758 [R38MB0AHCRN] Supports ABIs arm64-v8a, armeabi-v7a, armeabi
2020-07-30T09:02:33.068Z FTL/device 17758 [R38MB0AHCRN] Setup had an error Error: At least one of these paths should exist: /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/arm64-v8a/lib/android-30/minicap.so, /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/arm64-v8a/lib/android-30/minicap.so, /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/armeabi-v7a/lib/android-30/minicap.so, /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/armeabi-v7a/lib/android-30/minicap.so, /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/armeabi/lib/android-30/minicap.so, /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/armeabi/lib/android-30/minicap.so
    at Object.module.exports.requiredMatch (/home/bob/workspace/STF/lib/util/pathutil.js:33:9)
    at /home/bob/workspace/STF/lib/units/device/resources/minicap.js:41:23
    at SerialSyrup.ParallelSyrup.invoke (/home/bob/workspace/STF/node_modules/stf-syrup/lib/parallel.js:54:24)
    at /home/bob/workspace/STF/node_modules/stf-syrup/lib/serial.js:43:33
    at tryCatch1 (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/util.js:63:19)
    at Promise$_callHandler [as _callHandler] (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/promise.js:708:13)
    at Promise$_settlePromiseFromHandler [as _settlePromiseFromHandler] (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/promise.js:724:18)
    at Promise$_settlePromiseAt [as _settlePromiseAt] (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/promise.js:896:14)
    at Promise$_fulfillPromises [as _fulfillPromises] (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/promise.js:1041:14)
    at Async$_consumeFunctionBuffer [as _consumeFunctionBuffer] (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/async.js:74:12)
    at Async$consumeFunctionBuffer (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/async.js:37:14)
    at _combinedTickCallback (internal/process/next_tick.js:132:7)
    at process._tickCallback (internal/process/next_tick.js:181:9)
2020-07-30T09:02:33.070Z FTL/util:lifecycle 17758 [R38MB0AHCRN] Shutting down due to fatal error
2020-07-30T09:02:33.079Z INF/provider 8400 [*] Cleaning up device worker "R38MB0AHCRN"
2020-07-30T09:02:33.079Z ERR/provider 8400 [*] Device worker "R38MB0AHCRN" died with code 1
2020-07-30T09:02:33.080Z INF/provider 8400 [*] Restarting device worker "R38MB0AHCRN"
^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B^[[B2020-07-30T09:02:34.050Z INF/device:support:push 17778 [R38MB0AHCRN] Sending output to "tcp://127.0.0.1:7116"
2020-07-30T09:02:34.056Z INF/device 17778 [R38MB0AHCRN] Preparing device
2020-07-30T09:02:34.640Z INF/device:support:sub 17778 [R38MB0AHCRN] Receiving input from "tcp://127.0.0.1:7114"
2020-07-30T09:02:34.642Z INF/device:support:sub 17778 [R38MB0AHCRN] Subscribing to permanent channel "*ALL"
2020-07-30T09:02:34.711Z INF/device:support:properties 17778 [R38MB0AHCRN] Loading properties
2020-07-30T09:02:34.753Z INF/device:support:sdk 17778 [R38MB0AHCRN] Supports SDK 30
2020-07-30T09:02:34.754Z INF/device:support:abi 17778 [R38MB0AHCRN] Supports ABIs arm64-v8a, armeabi-v7a, armeabi
2020-07-30T09:02:34.757Z FTL/device 17778 [R38MB0AHCRN] Setup had an error Error: At least one of these paths should exist: /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/arm64-v8a/lib/android-30/minicap.so, /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/arm64-v8a/lib/android-30/minicap.so, /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/armeabi-v7a/lib/android-30/minicap.so, /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/armeabi-v7a/lib/android-30/minicap.so, /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/armeabi/lib/android-30/minicap.so, /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/armeabi/lib/android-30/minicap.so
    at Object.module.exports.requiredMatch (/home/bob/workspace/STF/lib/util/pathutil.js:33:9)
    at /home/bob/workspace/STF/lib/units/device/resources/minicap.js:41:23
    at SerialSyrup.ParallelSyrup.invoke (/home/bob/workspace/STF/node_modules/stf-syrup/lib/parallel.js:54:24)
    at /home/bob/workspace/STF/node_modules/stf-syrup/lib/serial.js:43:33
    at tryCatch1 (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/util.js:63:19)
    at Promise$_callHandler [as _callHandler] (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/promise.js:708:13)
    at Promise$_settlePromiseFromHandler [as _settlePromiseFromHandler] (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/promise.js:724:18)
    at Promise$_settlePromiseAt [as _settlePromiseAt] (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/promise.js:896:14)
    at Promise$_fulfillPromises [as _fulfillPromises] (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/promise.js:1041:14)
    at Async$_consumeFunctionBuffer [as _consumeFunctionBuffer] (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/async.js:74:12)
    at Async$consumeFunctionBuffer (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/async.js:37:14)
    at _combinedTickCallback (internal/process/next_tick.js:132:7)
    at process._tickCallback (internal/process/next_tick.js:181:9)
2020-07-30T09:02:34.758Z FTL/util:lifecycle 17778 [R38MB0AHCRN] Shutting down due to fatal error
2020-07-30T09:02:34.766Z INF/provider 8400 [*] Cleaning up device worker "R38MB0AHCRN"
2020-07-30T09:02:34.766Z ERR/provider 8400 [*] Device worker "R38MB0AHCRN" died with code 1
2020-07-30T09:02:34.767Z INF/provider 8400 [*] Restarting device worker "R38MB0AHCRN"
2020-07-30T09:02:35.696Z INF/device:support:push 17798 [R38MB0AHCRN] Sending output to "tcp://127.0.0.1:7116"
2020-07-30T09:02:35.701Z INF/device 17798 [R38MB0AHCRN] Preparing device
2020-07-30T09:02:36.337Z INF/device:support:sub 17798 [R38MB0AHCRN] Receiving input from "tcp://127.0.0.1:7114"
2020-07-30T09:02:36.338Z INF/device:support:sub 17798 [R38MB0AHCRN] Subscribing to permanent channel "*ALL"
2020-07-30T09:02:36.405Z INF/device:support:properties 17798 [R38MB0AHCRN] Loading properties
2020-07-30T09:02:36.451Z INF/device:support:sdk 17798 [R38MB0AHCRN] Supports SDK 30
2020-07-30T09:02:36.452Z INF/device:support:abi 17798 [R38MB0AHCRN] Supports ABIs arm64-v8a, armeabi-v7a, armeabi
2020-07-30T09:02:36.455Z FTL/device 17798 [R38MB0AHCRN] Setup had an error Error: At least one of these paths should exist: /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/arm64-v8a/lib/android-30/minicap.so, /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/arm64-v8a/lib/android-30/minicap.so, /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/armeabi-v7a/lib/android-30/minicap.so, /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/armeabi-v7a/lib/android-30/minicap.so, /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/armeabi/lib/android-30/minicap.so, /home/bob/workspace/STF/node_modules/minicap-prebuilt-beta/prebuilt/armeabi/lib/android-30/minicap.so
    at Object.module.exports.requiredMatch (/home/bob/workspace/STF/lib/util/pathutil.js:33:9)
    at /home/bob/workspace/STF/lib/units/device/resources/minicap.js:41:23
    at SerialSyrup.ParallelSyrup.invoke (/home/bob/workspace/STF/node_modules/stf-syrup/lib/parallel.js:54:24)
    at /home/bob/workspace/STF/node_modules/stf-syrup/lib/serial.js:43:33
    at tryCatch1 (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/util.js:63:19)
    at Promise$_callHandler [as _callHandler] (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/promise.js:708:13)
    at Promise$_settlePromiseFromHandler [as _settlePromiseFromHandler] (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/promise.js:724:18)
    at Promise$_settlePromiseAt [as _settlePromiseAt] (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/promise.js:896:14)
    at Promise$_fulfillPromises [as _fulfillPromises] (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/promise.js:1041:14)
    at Async$_consumeFunctionBuffer [as _consumeFunctionBuffer] (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/async.js:74:12)
    at Async$consumeFunctionBuffer (/home/bob/workspace/STF/node_modules/stf-syrup/node_modules/bluebird/js/main/async.js:37:14)
    at _combinedTickCallback (internal/process/next_tick.js:132:7)
    at process._tickCallback (internal/process/next_tick.js:181:9)
2020-07-30T09:02:36.457Z FTL/util:lifecycle 17798 [R38MB0AHCRN] Shutting down due to fatal error
2020-07-30T09:02:36.465Z INF/provider 8400 [*] Cleaning up device worker "R38MB0AHCRN"
2020-07-30T09:02:36.465Z ERR/provider 8400 [*] Device worker "R38MB0AHCRN" died with code 1
2020-07-30T09:02:36.465Z INF/provider 8400 [*] Restarting device worker "R38MB0AHCRN"
```

**Please run `stf doctor` and paste the output here.**
```
2020-07-30T09:28:45.340Z INF/cli:doctor 1640 [*] OS Arch: x64
2020-07-30T09:28:45.343Z INF/cli:doctor 1640 [*] OS Platform: linux
2020-07-30T09:28:45.343Z INF/cli:doctor 1640 [*] OS Platform: 4.4.0-174-generic
2020-07-30T09:28:45.343Z INF/cli:doctor 1640 [*] Using Node 8.16.1
2020-07-30T09:28:45.396Z INF/cli:doctor 1640 [*] Using ZeroMQ 4.1.4
2020-07-30T09:28:45.417Z ERR/cli:doctor 1640 [*] ProtoBuf is not installed (`protoc` is missing)
2020-07-30T09:28:45.419Z INF/cli:doctor 1640 [*] Using ADB 1.0.41
2020-07-30T09:28:46.226Z INF/cli:doctor 1640 [*] Using GraphicsMagick 1.3.23
2020-07-30T09:28:46.264Z INF/cli:doctor 1640 [*] Using RethinkDB 2.4.0~0xenial
```

**Comment by pcrepieux** [comment]
That should be solved now that stf uses minicap version 2.5.0

**Comment by Babu-SKK** [comment]
Yes. It is working now with minicap v2.5.0. Thanks
